main.tf

```diff
@@ -0,0 +1,16 @@
+locals {
+  records = { for r in var.records : "${r.type}_${r.name}.${var.domain}" => r }
+}
+
+resource "cloudflare_record" "this" {
+  for_each = local.records
+
+  zone_id = data.cloudflare_zone.this.id
+  name    = "${each.value.name}.${var.domain}."
```

Comment:
Are you sure this is not simply:
```
name = each.value.name
```
The domain itself is already identified by the `zone_id`

Reply:
I think this can be defined in two ways. both are correct

Reply:
changed to `each.value.name`